actions/class.Runner.php

```diff
@@ -233,36 +254,27 @@ protected function getErrorResponse($e = null, $prevResponse = [])
 
     /**
      * Gets an HTTP response code
-     * @param Exception [$e] Optional exception from which extract the error context
-     * @return int
      */
-    protected function getErrorCode($e = null)
+    protected function getStatusCodeFromException(Exception $exception): int
     {
-        $code = 200;
-        if ($e) {
-            $code = 500;
-
-            switch (true) {
-                case $e instanceof CatEngineConnectivityException:
-                case $e instanceof CatEngineNotFoundException:
-                case $e instanceof QtiRunnerEmptyResponsesException:
-                case $e instanceof QtiRunnerClosedException:
-                case $e instanceof QtiRunnerPausedException:
-                    $code = 200;
-                    break;
-
-                case $e instanceof common_exception_NotImplemented:
-                case $e instanceof common_exception_NoImplementation:
-                case $e instanceof common_exception_Unauthorized:
-                    $code = 403;
-                    break;
-
-                case $e instanceof tao_models_classes_FileNotFoundException:
-                    $code = 404;
-                    break;
-            }
+        switch (get_class($exception)) {
```

Comment:
Maybe we can do the same at 218 line, refactor switch case construction

Reply:
Nice catch, I'd consider doing that in another moment just to avoid more changes here.